File: csharp/API/V2/V2Mapper.cs
using API.Abstractions;
using Domain;

namespace API.V2;

public class V2Mapper: IMapper<User,V2User>
{
    public static V2User Map(User value)
    {
        return new V2User(
            UserUri: new Uri("/user/" + Uri.EscapeDataString(value.Id.Id.ToString())),
            Email: value.Email,
            IsActive: value.IsActive,
            Name: new V2UserName(
                FirstName: value.FirstName,
                LastName: value.LastName),
            Roles: value.Roles.Select(MapRole).ToArray());
    }

    private static V2UserRole MapRole(UserRole arg)
    {
        switch (arg)
        {
            case UserRole.Administrator: return V2UserRole.ADM;
            case UserRole.Support: return V2UserRole.SUP;
            case UserRole.Normal: return V2UserRole.USR;
            default: throw new Exception(arg.ToString());
        }
    }
}

File: java/src/main/java/se/gewalli/users/model/V2Mapper.java
package se.gewalli.users.model;

import se.gewalli.users.domain.User;
import se.gewalli.users.domain.UserRole;

import java.util.stream.Collectors;

public class V2Mapper {
    public static V2User map(User value)
    {
        return new V2User().userUri("/users/" + (value.getId().getId()))
                .email(value.getEmail())
                .isActive(value.isActive())
                .name(new V2UserName().firstName(value.getFirstName()).lastName(value.getLastName()))
                .roles(value.getRoles().stream().map(V2Mapper::mapRole).collect(Collectors.toList()));
    }

    private static V2UserRole mapRole(UserRole arg)
    {
        switch (arg)
        {
            case Administrator: return V2UserRole.Administrator;
            case Support: return V2UserRole.Support;
            case Normal: return V2UserRole.Normal;
            default: throw new RuntimeException(arg.toString());
        }
    }
}
